editor/src/components/canvas/canvas-strategies/shared-absolute-resize-strategy-helpers.ts

```diff
@@ -79,3 +88,175 @@ function pinsForEdgePosition(edgePosition: EdgePosition): AbsolutePin[] {
 
   return [...horizontalPins, ...verticalPins]
 }
+
+type IsCenterBased = 'center-based' | 'non-center-based'
+
+export function resizeBoundingBox(
+  boundingBox: CanvasRectangle,
+  drag: CanvasPoint,
+  edgePosition: EdgePosition,
+  lockedAspectRatio: number | null,
+  centerBased: IsCenterBased,
+): CanvasRectangle {
+  let dragToUse = drag
+  let cornerEdgePosition = edgePosition
+  let startingCornerPosition = {
+    x: 1 - edgePosition.x,
+    y: 1 - edgePosition.y,
+  } as EdgePosition
+
+  const isEdgeHorizontalSide = isEdgePositionAHorizontalEdge(edgePosition)
+  const isEdgeVerticalSide = isEdgePositionAVerticalEdge(edgePosition)
+
+  if (isEdgeHorizontalSide) {
+    dragToUse = canvasPoint({ x: 0, y: drag.y })
+    startingCornerPosition = { x: 0, y: startingCornerPosition.y }
+    cornerEdgePosition = { x: 1, y: edgePosition.y }
+  } else if (isEdgeVerticalSide) {
+    dragToUse = canvasPoint({ x: drag.x, y: 0 })
+    startingCornerPosition = { x: startingCornerPosition.x, y: 0 }
+    cornerEdgePosition = { x: edgePosition.x, y: 1 }
+  }
+
+  const oppositeCornerPoint = pickPointOnRect(boundingBox, startingCornerPosition)
+  const draggedCorner = pickPointOnRect(boundingBox, cornerEdgePosition)
+  let newCorner = offsetPoint(draggedCorner, dragToUse)
+
+  if (lockedAspectRatio != null) {
+    const horizontalLineB = canvasPoint({
+      x: newCorner.x + 100,
```

Comment:
what's this magic +100 here?

is it a problem that you maybe reassign newCorner at line 136?

Reply:
This just creates a horizonal and a vertical line here. It doesn't matter if this is 100 or a different constant, what matters is just that the other coordinate remains the same.
Line 136 is ok, that is what calculates the horizontal/vertical projection of this to the diagonal (which will the good newCorner)